### ISSUE TITLE
Invalid model generated

### ISSUE THREAD

**Issue by bvwells** [go-swagger/go-swagger#1653]
## Problem statement

Invalid code is generated when the swagger spec below is run through go-swagger to generate a client.
The following code is generated which fails to compile:

```
func (m *Pets) validateCrazyPets(formats strfmt.Registry) error {

	if err := validate.Required("CrazyPets", "body", m.CrazyPets); err != nil {
		return err
	}

	for k := range m.CrazyPets {

	}

	return nil
}
```

## Swagger specification

```
swagger: '2.0'
info:
  version: v1
  title: Model bug
  description: Swagger definition showing bug in go-swagger model generation
basePath: /SomeBasePath
schemes:
  - http
paths:
  /api/pets:
    get:
      tags:
        - Pets
      summary: Get all pets
      consumes:
        - application/json
      produces:
        - application/json
      responses:
        '200':
          description: Provides the pets in some crazy data structure.
          schema:
            $ref: '#/definitions/Pets'
definitions:
  Pets:
    description: Some crazy pets data structure.
    required:
      - CrazyPets
    type: object
    properties:
      CrazyPets:
        description: Crazy pets
        type: object
        additionalProperties:
          type: object
          additionalProperties:
            type: array
            items:
              type: string
```

## Steps to reproduce
- Run go-swagger with following command: `swagger generate client`
- Try to build code to get compilation error in models\pets.go. 
```./pets.go:45:6: k declared and not used```

## Environment
swagger version: version: 0.14.0 commit: 25e637c5028dee7baf8cdf5d172ccb28cb8e5c3e
go version: go1.10.1 linux/amd64
OS:  Linux

**Comment by fredbi** [comment]
Can you reproduce this with current master?

**Comment by fredbi** [comment]
Your spec generates and builds properly in 0.15.

**Comment by bvwells** [comment]
After a bit of digging I guess it might have been fixed in PR #1521...
